dattri/benchmark/load.py

```diff
@@ -0,0 +1,268 @@
+
+
+def generate_url_map(identifier: str) -> Dict[str, Any]:
+    """Generate the URL map for the benchmark setting.
+
+    Args:
+        identifier (str): The identifier for the benchmark setting.
+
+    Returns:
+        Dict[str, Any]: The URL map for the benchmark setting.
+    """
+    return {
+        "models_full": REPO_URL + f"{identifier}/{identifier}_full.zip?download=true",
+        "models_half": REPO_URL + f"{identifier}/{identifier}_half.zip?download=true",
+        "groundtruth": {
+            "lds": [
+                REPO_URL + f"{identifier}/lds/indices_lds.pt?download=true",
+                REPO_URL + f"{identifier}/lds/target_values_lds.pt?download=true",
+            ],
+            "loo": [
+                REPO_URL + f"{identifier}/loo/indices_loo.pt?download=true",
+                REPO_URL + f"{identifier}/loo/target_values_loo.pt?download=true",
+            ],
+        },
+    }
+
+
+SUPPORTED_DATASETS = {
+    "mnist": create_mnist_dataset,
+}
+
+LOSS_MAP = {
+    "mnist_mlp": loss_mnist_mlp,
+    "mnist_lr": loss_mnist_lr,
+}
+
+TRAIN_FUNC_MAP = {
+    "mnist_mlp": train_mnist_mlp,
+    "mnist_lr": train_mnist_lr,
+}
+
+MODEL_MAP = {
+    "mnist_mlp": partial(create_mlp_model, "mnist"),
+    "mnist_lr": partial(create_lr_model, "mnist"),
+}
+
+
+def _count_folders(directory_path: str) -> int:
+    """Count the number of folders in a directory.
+
+    Args:
+        directory_path (str): The path to the directory.
+
+    Returns:
+        int: The number of folders in the directory.
+    """
+    items = os.listdir(directory_path)
+    folders = [item for item in items if (directory_path / item).is_dir()]
+    return len(folders)
+
+
+def _download(
+    url: str,
+    destination_path: str = ".",
+    unzip: bool = False,
+    file_name: str = "file.zip",
+) -> None:
+    """Download helper function.
+
+    Args:
+        url (str): The URL of the file to download.
+        destination_path (str): The path to save the downloaded file.
+        unzip (bool): Whether to unzip the downloaded file.
+        file_name (str): The name of the downloaded file.
+
+    Raises:
+        ValueError: If the download fails.
+    """
+    # Send a GET request to the URL
+    response = requests.get(url)  # noqa: S113
+
+    # Check if the request was successful
+    destination_path.mkdir(parents=True, exist_ok=True)
+    if response.status_code == 200:  # noqa: PLR2004
+        if unzip:
+            with zipfile.ZipFile(BytesIO(response.content)) as zip_ref:
+                zip_ref.extractall(destination_path)
+        else:
+            with (destination_path / file_name).open(mode="wb") as file:
+                for chunk in response.iter_content(chunk_size=1024):
+                    if chunk:
+                        file.write(chunk)
+    else:
+        error_msg = f"Failed to download file. Status code: {response.status_code}"
+        raise ValueError(error_msg)
+
+
+def load_benchmark(
```

Comment:
do we need a test file or script for this function?

Reply:
yes, added one and verify the performance.